web/docs/tql2/operators/summarize.md

```diff
@@ -63,6 +63,9 @@ differently, take exactly one argument:
 - `count`: When used as `count()`, simply counts the events in the group. When
   used as `count(x)`, counts all grouped values that are not null.
 - `count_distinct`: Counts all distinct grouped values that are not null.
+- `over`: Given `fun(…).over(str)`, computes the aggregation function `fun(…)`
+  per unique non-null value of `str`, returning the record `{str: fun(…), …}`
+  with the values of `str` lifted into field names of the returned record.
```

Comment:
As I mentioned in Slack, I'm not yet sure whether this is the right approach to the problem, but after playing with it for a few hours I think it works really well. I figured I'd just open the PR to continue the discussion and to not forget about this.